clang/lib/Sema/SemaInit.cpp

```diff
@@ -8194,25 +8216,18 @@ void Sema::checkInitializerLifetime(const InitializedEntity &Entity,
       }
 
       switch (shouldLifetimeExtendThroughPath(Path)) {
+      case PathLifetimeKind::ShouldExtend:
```

Comment:
Do we need separate `ShouldExtend` and `Extend` kinds still, or can we combine them?